frontend/model/contracts/group.js

```diff
@@ -1083,6 +1084,9 @@ sbp('chelonia/defineContract', {
             case 'nonMonetaryEdit':
               nonMonetary.splice(nonMonetary.indexOf(value.replace), 1, value.with)
               break
+            case 'nonMonetaryReplace':
+              groupProfile.nonMonetaryContributions = cloneDeep(value)
+              break
```

Comment:
Adding this feature requires a way to update `groupProfile.nonMonetaryContributions` array as a whole instead of element by element. So adding `nonMonetaryReplace` here.